### PR TITLE
Add maven test

### PR DESCRIPTION
**Pull request by Or-Geva**
- [x] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/fileutils.go [hunk 1]
```diff
@@ -21,8 +21,11 @@ const (
 	maxFileAge = 24.0
 )
 
-func IsFileExists(path string) (bool, error) {
-	fileInfo, err := os.Stat(path)
+// Check if path points at a file.
+// If path points at a symlink and `preserveSymLink == true`,
+// function will return `true` regardless of the symlink target
+func IsFileExists(path string, preserveSymLink bool) (bool, error) {
```
Comment (yahavi) [inline utils/fileutils.go:27]: If I understand correctly, it is not "preserve", but more like "follow", not? It is a read operation.
```suggestion
func IsFileExists(path string, followSymlink bool) (bool, error) {
```

File: utils/fileutils.go [hunk 2]
```diff
@@ -196,3 +310,134 @@ func FindFileInDirAndParents(dirPath, fileName string) (string, error) {
...
+	if err != nil {
+		return err
+	}
+	dstFile, err := os.Create(dstPath)
+	if err != nil {
+		return err
+	}
+	defer dstFile.Close()
+	io.Copy(dstFile, srcFile)
+	return nil
```
Comment (yahavi) [inline utils/fileutils.go:372]:
```suggestion
	_, err = io.Copy(dstFile, srcFile)
	return err
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: entities/buildinfo.go [hunk 3]
```diff
@@ -310,8 +319,17 @@ type Dependency struct {
 	*Checksum
 }
 
-func (d *Dependency) IsEqual(b Dependency) bool {
-	return d.Id == b.Id && d.Type == b.Type && d.Sha1 == b.Sha1 && d.Md5 == b.Md5
+func (d *Dependency) IsEqual(a Dependency) bool {
```
Comment (yahavi) [inline entities/buildinfo.go:322]:
```suggestion
func (d *Dependency) IsEqual(other Dependency) bool {
```

File: build/golang_test.go [hunk 4]
```diff
@@ -23,3 +26,22 @@ func validateModule(t *testing.T, module buildinfo.Module, expectedDependencies,
...
+	goBuild, err := service.GetOrCreateBuild("build-info-go-test-golang", "1")
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, goBuild.Clean())
+	}()
+	goModule, err := goBuild.AddGoModule(filepath.Join("testdata", "golang", "project"))
+	assert.NoError(t, err)
+	err = goModule.CalcDependencies()
+	assert.NoError(t, err)
+	err = goModule.AddArtifacts(entities.Artifact{Name: "artifactName", Type: "artifactType", Path: "artifactPath", Checksum: &entities.Checksum{Sha1: "123", Md5: "456", Sha256: "789"}})
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply build/golang_test.go:41]: I think we do check it in validateModule

**Pull request merged**